lib/fog/vsphere/models/compute/snapshot.rb

```diff
@@ -0,0 +1,42 @@
+require 'fog/compute/models/server'
+
+module Fog
+  module Compute
+    class Vsphere
+      class Snapshot < Fog::Model
+
+        identity  :ref
+        attribute :server_id
+
+        attribute :name
+        attribute :quiescedi, :default => false
+        attribute :description, :default => ''
+        attribute :create_time
+        attribute :power_state, :default => 'none'
+        attribute :ref
+        attribute :mo_ref
+        attribute :tree_node
+        attribute :snapshot_name_chain
+        attribute :ref_chain
+
+        attr_reader :service
+
+        def initialize(attributes = {})
+          super attributes
+        end
+
+        def child_snapshots(filters = {})
+          requires :mo_ref
+          service.snapshots(
+            { :server_id => server_id, :parent_snapshot => self }.update(filters)
+          )
+        end
+
+        def get(snapshot_ref)
```

Comment:
Hmmm. get_child sound nice.

Em qua, 4 de nov de 2015 12:34, ireznice <notifications@github.com>
escreveu:

> In lib/fog/vsphere/models/compute/snapshot.rb
> <https://github.com/fog/fog/pull/3749#discussion_r43886709>:
>
> > +        attribute :ref_chain
> > +
> > +        attr_reader :service
> > +
> > +        def initialize(attributes = {})
> > +          super attributes
> > +        end
> > +
> > +        def child_snapshots(filters = {})
> > +          requires :mo_ref
> > +          service.snapshots(
> > +            { :server_id => server_id, :parent_snapshot => self }.update(filters)
> > +          )
> > +        end
> > +
> > +        def get(snapshot_ref)
>
> Do you have some idea? It must be singular because it returns only self or
> some child snapshot. Do you think that get_child is OK?
>
> Thanks.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fog/fog/pull/3749/files#r43886709>.
>